tests/test_cli.py

```diff
@@ -0,0 +1,336 @@
+    return status_mock_helper(mocker, gcloud_mock, None)
+
+
+def status_mock_helper(mocker, gcloud_mock, run_name):
+    setup = mocker.patch("mlbench_core.cli.cli.setup_client_from_config")
+    setup.return_value = True
+
+    client = gcloud_mock["apiclient"]
+    rundict = {
+        "name": "none" if run_name is None else run_name,
+        "job_id": "",
+        "job_metadata": "",
+        "id": "my-id",
+    }
+
+    client.return_value.get_runs.return_value.result.return_value.json.return_value = [
+        rundict
+    ]
+
+    return client, rundict["id"], rundict["name"]
+
+
+def create_gcloud_test_helper(mocker, gcloud_mock, args, option_dict=None):
+    container_v1 = gcloud_mock["containerv1"]
+    tiller = gcloud_mock["tiller"]
+    auth = gcloud_mock["auth"]
+
+    cluster = container_v1.types.Cluster
+    get_operation = container_v1.ClusterManagerClient.return_value.get_operation
+    nodeconfig = container_v1.types.NodeConfig
+    accelerator = container_v1.types.AcceleratorConfig
+
+    install_release = tiller.return_value.install_release
+
+    if option_dict is not None and "project" in option_dict:
+        project = option_dict["project"]
+    else:
+        _, project = auth()
+
+    cmd_line = get_gcloud_cmd_line(args, option_dict)
+
+    if option_dict is None:
+        option_dict = CREATE_GCLOUD_DEFAULTS
+    else:
+        for k in CREATE_GCLOUD_DEFAULTS:  # add missing defaults
+            if k not in option_dict:
+                option_dict[k] = CREATE_GCLOUD_DEFAULTS[k]
+
+    create_gcloud(cmd_line)
+
+    cluster.assert_called_once()
+    assert cluster.call_args[1]["initial_node_count"] == args[0]  # num_workers
+
+    nodeconfig.assert_called_once()
+    assert nodeconfig.call_args[1]["machine_type"] == option_dict["machine_type"]
+    assert nodeconfig.call_args[1]["disk_size_gb"] == option_dict["disk_size"]
+    assert nodeconfig.call_args[1]["preemptible"] == option_dict["preemptible"]
+
+    tiller.assert_called_once()
+    install_release.assert_called_once()
+
+    assert install_release.call_args[1]["values"]["limits"]["workers"] == args[0] - 1
+    assert (
+        install_release.call_args[1]["values"]["limits"]["gpu"]
+        == option_dict["num_gpus"]
+    )
+    assert (
+        install_release.call_args[1]["values"]["limits"]["cpu"]
+        == option_dict["num_cpus"]
+    )
+
+    get_operation.assert_called()
+    assert option_dict["zone"] in get_operation.call_args[1]["name"]
+    assert project in get_operation.call_args[1]["name"]
+
+    if option_dict["num_gpus"] > 0:
+        num_gpus = option_dict["num_gpus"]
+        gpu_type = option_dict["gpu_type"]
+        accelerator.assert_called_once_with(
+            accelerator_count=num_gpus, accelerator_type=gpu_type
+        )
+
+
+def test_invalid_num_workers(mocker, gcloud_auth):
+
+    with pytest.raises(AssertionError):
+        create_gcloud(["1", "test"])
+
+
+def test_create_cluster_default(mocker, gcloud_mock):
```

Comment:
`test_create_cluster_default`, `test_create_cluster_non_default` and `test_create_cluster_gpu` can be combined into a single test by parametrizing them https://docs.pytest.org/en/latest/parametrize.html